nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/preliminary/MachineMemoryUsageHealthCheck.java

```diff
@@ -186,31 +125,115 @@ private HealthCheckResult doCheckNonLinux() {
         return result;
     }
 
-    private static boolean isLinux() {
-        String osName = System.getProperty("os.name");
-        return osName.startsWith("Linux") ||
-                osName.startsWith("FreeBSD") ||
-                osName.startsWith("OpenBSD") ||
-                osName.startsWith("gnu") ||
-                osName.startsWith("netbsd");
+    @Override
+    public void collect(MonitoringWatchCollector collector) {
+        collectUsage(collector, "ns:health PhysicalMemoryUsage", "Physical Memory Usage", 5, false);
     }
 
-    private static long parseMemInfo(String[] parts) {
-        long memory = 0;
-
-        if (parts.length >= 2) {
-            memory = Long.parseLong(parts[1]);
-            if (parts.length > 2 && KB.equals(parts[2])) {
-                memory *= 1024;
+    @Override
+    @MonitoringData(ns = "health", intervalSeconds = 12)
+    public void collect(MonitoringDataCollector collector) {
+        if (getOptions().isEnabled()) {
+            try {
+                collector.collect("PhysicalMemoryUsage", (long) stats.usedPercentage());
+            } catch (Exception ex) {
+                throw ex instanceof RuntimeException ? (RuntimeException) ex : new RuntimeException(ex);
             }
         }
-        return memory;
     }
 
-    private static Long invokeMethodFor(OperatingSystemMXBean osBean, String methodName) throws Exception {
-        Method m = osBean.getClass().getDeclaredMethod(methodName);
-        m.setAccessible(true);
-        return (Long) m.invoke(osBean);
+    private static final class PysicalMemoryUsage {
+        private volatile long totalMemory;
+        private volatile long availableMemory;
+
+        PysicalMemoryUsage() {
+            // make visible
+        }
+
+        long getTotalMemory() {
+            return totalMemory;
+        }
+
+        long getUsedMemory() {
+            return totalMemory - availableMemory;
+        }
+
+        double usedPercentage() throws Exception {
+            if (isLinux()) {
+                updateLinux();
+            } else {
+                updateNonLinux();
+            }
+            long usedMemory = totalMemory - availableMemory;
+            return totalMemory == 0L ? 0d : 100d * usedMemory / totalMemory;
+        }
+
+        private void updateLinux() throws IOException {
+            List<String> lines = Files.readAllLines(Paths.get("/proc/meminfo"), StandardCharsets.UTF_8);
+            if (lines.isEmpty()) {
+                totalMemory = 0;
+                availableMemory = 0;
+                return;
+            }
+            long otherAvailableMemory = 0;
```

Comment:
NB: I introduced `otherAvailableMemory` as single variable as the individual variables I removed would only be used to summarise them.